### PR TITLE
Cosmos ARM changes for Managed Identity Authentication

### PR DESCRIPTION
**Pull request by v-ajajvanu**
## Description
Made changes in OSS deployment ARM Template to create Cosmos DB instance with Managed Identity Authentication support.

## Related issues
Addresses [[issue #122541](https://microsofthealth.visualstudio.com/Health/_workitems/edit/122541)].

## Testing
Made the ARM template changes and created new OSS Gen 1 instance using the modified ARM template.

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- Tag the PR with **Schema Version backward compatible** or **Schema Version backward incompatible** or **Schema Version unchanged** if this adds or updates Sql script which is/is not backward compatible with the code.
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: samples/templates/default-azuredeploy-docker.json [hunk 1]
```diff
@@ -530,7 +578,7 @@
                 "template": {
                     "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
                     "contentVersion": "1.0.0.0",
-                   
+
```
Comment (abiisnn) [inline samples/templates/default-azuredeploy-docker.json:581]: NIT: empty line

**Review by abiisnn — APPROVED** [review]
🍒

File: samples/templates/default-azuredeploy-docker.json (hunk 1 above)
Comment (v-ajajvanu) [reply samples/templates/default-azuredeploy-docker.json]: @abiisnn I had made the changes to remove the additional blank line. Changes are committed into this PR.

**Comment by brendankowitz** [comment]
Covered by changes in #4026

**Pull request closed without merging**